### ISSUE TITLE
Issue with Nexus generate _self

### ISSUE THREAD

**Issue by MFSY** [BlueBrain/nexus#686]
When I create a resource https://bbp.epfl.ch/neurosciencegraph/data/persons/uuid against the following endpoint:

schema: https://bbp.epfl.ch/nexus/v1/resources/bbp/agents/datashapes:person

The generated _self is: 

"_self": "https://bbp.epfl.ch/nexus/v1/resources/bbp/agents/datashapes:person/persons/uuid"

which returns (404):

```
{
    "@context": "https://bluebrain.github.io/nexus/contexts/error.json",
    "@type": "NotFound",
    "reason": "The requested resource could not be found."
}
```

The projet configuration is :

```
{
    "@context": [
        "https://bluebrain.github.io/nexus/contexts/admin.json",
        "https://bluebrain.github.io/nexus/contexts/resource.json"
    ],
    "@id": "https://bbp.epfl.ch/nexus/v1/projects/bbp/agents",
    "@type": "Project",
    "description": "This project stores agents (persons,...) data.",
    "base": "https://bbp.epfl.ch/neurosciencegraph/data/",
    "vocab": "https://bbp.epfl.ch/nexus/v1/resources/bbp/agents/_/",
    "apiMappings": [
        {
            "prefix": "datashapes",
            "namespace": "https://neuroshapes.org/dash/"
        },
        {
            "prefix": "prov",
            "namespace": "http://www.w3.org/ns/prov#"
        },
        {
            "prefix": "ontologies",
            "namespace": "https://neuroshapes.org/dash/ontology"
        },
        {
            "prefix": "context",
            "namespace": "https://incf.github.io/neuroshapes/contexts/"
        },
        {
            "prefix": "provcommonshapes",
            "namespace": "https://provshapes.org/commons/"
        },
        {
            "prefix": "commonshapes",
            "namespace": "https://neuroshapes.org/commons/"
        },
        {
            "prefix": "provdatashapes",
            "namespace": "https://provshapes.org/datashapes/"
        },
        {
            "prefix": "taxonomies",
            "namespace": "https://neuroshapes.org/dash/taxonomy"
        },
        {
            "prefix": "schemaorg",
            "namespace": "http://schema.org/"
        }
    ]
}
```

**Comment by umbreak** [comment]
It seems to be an issue which has been fixed in v1.1
I tried to reproduce it in dev and the self gets generated correctly. Please try and let me know if you still get the same error.

**Comment by umbreak** [comment]
I reopen the ticket. I found the situation under which this happens. Is when an @id has the following value: `{project_base}/some_intermidiate_folder/uuid`
